### DISCUSSION TITLE
Static site

### DISCUSSION THREAD

**Discussion by sebazelonka** [shuding/nextra#1853] [Q&A]
Hello, I'm newbie with Next and also Nextra. 

I want to build a full static site to deploy it into gitlab pages, is that possible?

I need to build locally, because there are some permission issues with some NPM packages, so I need to manage all that locally, and then push the build content to be published. Basically, what I need is a simple `Public` folder with HTMLs, CSS and JS

I've tried adding `output: 'export',` but didn't work

```
const withNextra = require('nextra')({
  theme: 'nextra-theme-docs',
  themeConfig: './theme.config.tsx',
  output: 'export'
})
```

Any help is welcomed :) 

Thanks,
Seba

**Comment by dimaMachina** [comment]
Nextra can be built as full static website
More info https://nextjs.org/docs/pages/building-your-application/deploying/static-exports

**Reply by sebazelonka** [reply]
Could you add extra info on your answer?

As I said in my question, I've tried to include `output: 'export'` on nextra config object, but didn't work.

**Reply by stelloprint** [reply]
I have the same issue running builds. The console suggests the builds are taking place however an `out` folder is nowhere to be found even after running this `find . -type d | grep "/out/"` once the process completes.

<img width="719" alt="failedBuilds" src="https://github.com/shuding/nextra/assets/58522233/1ce3f875-8673-4cd4-b256-e0953bb5a4b9">

this what the next.config.js file looks like:

```js
// this handles basePath for <Link/> and assetPrefix for <Image/> components exported statically to GH Pages: 
// https://www.viget.com/articles/host-build-and-deploy-next-js-projects-on-github-pages/
const isGithubActions = process.env.GITHUB_ACTIONS || false;
let assetPrefix = '';
let basePath = '';


if (isGithubActions) {
  // trim off `<owner>/`
  const repo = process.env.GITHUB_REPOSITORY.replace(/.*?\//, '');
  assetPrefix = `/${repo}/`;
  basePath = `/${repo}`;
}

// initializes the withNextra config object with theme settings
const withNextra = require('nextra')({
  theme: 'nextra-theme-docs',
  themeConfig: './theme.config.tsx',
});

// exports the config with some additional nextjs settings
module.exports = withNextra({
  assetPrefix: assetPrefix,
  basePath: basePath,
  output: 'export',
  images: {
    domains: ['xyz.imgix.net'],
    loader: 'imgix',
    loaderFile: './components/imgix/imgix.js',
    path: 'xyz.imgix.net',
  },
});
```

**Reply by stelloprint** [reply]
the above next.config.js file results in this error:

`Error: Specified images.loader property (imgix) cannot be used with images.loaderFile property. Please set images.loader to "custom".`

which does not make sense since nextjs docs say that's exactly what has to be done. 

EDIT: the docs only describe a 'custom' loader without mention of other options however the code base does list supported values for that property including imgix: https://github.com/vercel/next.js/blob/50dff9390268e402492db5a1bdc43ca781a94f43/packages/next/src/shared/lib/image-config.ts#L4

maybe this is a new error that has not been documented yet and my guide uses old syntax
